simple_graph.py

```diff
@@ -137,3 +137,61 @@ def __getitem__(self, item):
             raise KeyError("Node not in graph")
 
         return item.edges
+
+    def dijkstra(self, source):
+
+        dist = {}
+        prev = {}
+
+        dist[source] = 0
+        prev[source] = None
+
+        dq = deque()
+
+        for node in self.nodes_:
+            if node != source:
+                dist[node] = float('Inf')
+                prev[node] = None
+            dq.append(node)
+
+        while dq:
+
+            sorted_dq = sorted(dq, key=dist.__getitem__)
```

Comment:
it's a bit odd to use a special method directly like this.  You could specify `dict.get` as the key instead.